src/services/session.service.ts

```diff
@@ -70,14 +68,14 @@ export class SessionService {
 
     removeWarningBadge()
 
-    return session
+    return memorySession
   }
 
-  private async encryptKey(privateKey: PrivateKey): Promise<KeyData> {
+  private async encryptSeed(seed: Uint8Array): Promise<KeyData<string>> {
     const sessionKey = generateRandomString(128)
     const url = this.generateRandomUrl()
 
-    const encryptedPrivateKey = AES.encrypt(privateKey, sessionKey).toString()
+    const encryptedSeed = aesEncryptBytes(seed, sessionKey)
```

Comment:
I cannot really comprehend with this part, we want to fetch the wallet from the localstorage because it is safer than store on any network and also not forcing users to create portable account. 
because of this, the previous approach with cookies cannot be used because at the end of the session the cookies will be purged.

Here, as I can see, you still operating with temporary random generated key with which you encrypt the private key seed. The problem with this is that, you don't have the encryption key after closing the browser and trying to login to your account again.

Can't we use the hash of the password as encryption key for seed encryption? this password hash could be kept in the sessionStorage of the Extension.